packages/archives.sls

```diff
@@ -26,7 +26,11 @@ packages-archive-unwanted-{{ file_or_directory }}:
 
   # wanted 'archive' software
   {%- for package, archive in wanted_archives.items() %}
-    {%- set archivename = archive.dl.source.split('/')[-1] %}
```

Comment:
I would suggest a change here, to make the destination name configurable through a variable (like `name`, `destname` or `binname`?) that will also make it backward compatible:

```sls
{%- set archivename = archive.dl.source.split('/')[-1] if 'destname' not in archive else archive.destname %}
```

This way, we can define an arbitrary destination name like

```yaml
packages:
  archives:
    wanted:
      yq:
        dest: /usr/local/bin
        destname: my_yq
          dl:
            format: bin
            source: https://github.com/mikefarah/yq/releases/download/v4.13.4/yq_linux_amd64
```

Another backward-compatible alternative is to add a boolean parameter like `destname_as_package: true` (or any better var name you can think of) and then

```sls
{%- set use_packagename_as_binaryname = archive.destname_as_package | default(false) %}
{%- set archivename = package if use_packagename_as_binaryname else archive.dl.source.split('/')[-1] %}
```
```yaml
packages:
  archives:
    wanted:
      yq:
        dest: /usr/local/bin
        archive.destname_as_package: true
          dl:
            format: bin
            source: https://github.com/mikefarah/yq/releases/download/v4.13.4/yq_linux_amd64
```
Although I think this 2nd option is less flexible that the 1st one.

wdyt?

Reply:
I think you're right and that's a good way to keep everyone happy.  Committed this, CI should be running soon.

Reply:
@mdschmitt ?